File: exercises/101-109_small_problems/easy_01/easy_05.rb
# easy_05.rb

def reverse_sentence(string)
  string.split.reverse.join(' ')
end

=begin
utp
- takes a string and returns it in reverse order
- need to account for an empty string

i/o/d
- i: string
- o: string
- d: string, largely

test cases:
puts reverse_sentence('') == ''
puts reverse_sentence('Hello World') == 'World Hello'
puts reverse_sentence('Reverse these words') == 'words these Reverse'

algorithim
- split string into array, reverse array, join string back together with ' ' as
  separator


=end

File: exercises/101-109_small_problems/easy_01/go/easy_05.go
package main

import (
	"fmt"
	"strings"
)

func reverseSentence(s string) string {
	words := strings.Split(s, " ")
	reversed := []string{}

	for i := len(words) - 1; i >= 0; i-- {
		reversed = append(reversed, words[i])
	}

	return strings.Join(reversed, " ")
}

func main() {
	fmt.Println(reverseSentence(""))
	fmt.Println(reverseSentence("Hello World"))
	fmt.Println(reverseSentence("Reverse these words"))
}
